src/lib/core/fiber_pool.c

```diff
@@ -61,7 +61,9 @@ fiber_pool_f(va_list ap)
 			f->caller->flags |= FIBER_IS_READY;
 			assert(f->caller->caller == &cord->sched);
 		}
+		fiber_set_system(fiber(), false);
```

Comment:
This means that all new fibers created by a pool should be system, but I don't see where they are marked so.

Also, there should be a comment somewhere explaining why pool fibers are marked system.

Reply:
Let's make intention explicit thru use of `fiber_new_system` for creating pool fiber. Also add a comment.

```diff
--- a/src/lib/core/fiber_pool.c
+++ b/src/lib/core/fiber_pool.c
@@ -133,7 +133,17 @@ fiber_pool_cb(ev_loop *loop, struct ev_watcher *watcher, int eve>
                        f = rlist_shift_entry(&pool->idle, struct fiber, state);
                        fiber_call(f);
                } else if (pool->size < pool->max_size) {
-                       f = fiber_new(cord_name(cord()), fiber_pool_f);
+                       /*
+                        * We don't want fibers to be cancellable by client
+                        * while they are in the pool. However system flag is
+                        * reset during proccessing message from pool endpoint
+                        * so that fiber is made cancellable back.
+                        *
+                        * If some message processing should not be cancellable
+                        * by client then it can just set system flag during
+                        * it's execution.
+                        */
+                       f = fiber_new_system(cord_name(cord()), fiber_pool_f);
                        if (f == NULL) {
                                diag_log();
                                break;
```